### PR TITLE
"make lint" work in Mac and fix lint issues

### LINKED ISSUE #1977: "make lint" needs to work in Mac
**What would you like to be added/modified**:
"make lint" currently does not work in Mac. It needs to work in Mac for better developer experience

**Why is this needed**:
Better developer experience,  as a lot of developers use Mac

### PR DESCRIPTION
**Pull request by dingyin**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
/kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test
> /kind feature

**What this PR does / why we need it**:
"make lint" work in Mac and fix lint issues

**Which issue(s) this PR fixes**:

Fixes #1977

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: hack/lib/lint.sh [hunk 1]
```diff
@@ -20,17 +20,51 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
+SED_CMD=""
+
+if [[ "$OSTYPE" == "darwin"* ]]
```
Comment (daixiang0) [inline hack/lib/lint.sh:25]: Better to use `switch`.

**Comment by kevin-wangzefeng** [comment]
I met the following issue on my mac, maybe due to golangci-lint version not up to date?
```console
make lint
hack/make-rules/lint.sh
ERRO Running error: no such linter "whitespace"
make: *** [lint] Error 3

```

**Comment by daixiang0** [comment]
Yep, i recommand that move lint test into container then we can run soomthly at any OS and ensure all dependences.

**Comment by dingyin** [comment]
> I met the following issue on my mac, maybe due to golangci-lint version not up to date?
> 
> ```
> make lint
> hack/make-rules/lint.sh
> ERRO Running error: no such linter "whitespace"
> make: *** [lint] Error 3
> ```
I have this version.
golangci-lint has version 1.29.0 built from 6a68907 on 2020-07-20T14:54:31Z

**Comment by dingyin** [comment]
> I met the following issue on my mac, maybe due to golangci-lint version not up to date?
> 
> ```
> make lint
> hack/make-rules/lint.sh
> ERRO Running error: no such linter "whitespace"
> make: *** [lint] Error 3
> ```

Successfully repro your issue. You shouldn't have upstaged changes. All changes have to be committed or staged.

File: cloud/pkg/csidriver/controllerserver.go [hunk 2]
```diff
@@ -20,8 +20,6 @@ import (
 	"encoding/base64"
 	"encoding/json"
 	"errors"
-	"fmt"
-
```
Comment (fisherxu) [inline cloud/pkg/csidriver/controllerserver.go:24]: leave the empty line here :)

Comment (dingyin) [reply cloud/pkg/csidriver/controllerserver.go:24]: addressed

**Pull request merged**